src/backend/marsha/core/models/video.py

```diff
@@ -245,6 +262,91 @@ def get_mediapackage_endpoints(self):
         """Return the mediapackage enspoints info."""
         return self.live_info.get("mediapackage").get("endpoints")
 
+    @property
+    def is_recording(self):
+        """Whether the video is currently recording."""
+        if not self.recording_slices:
+            return False
+        last_recording_slice = self.recording_slices[-1]
+        return (
+            last_recording_slice.get("start")
+            and last_recording_slice.get("stop") is None
+        )
+
+    @property
+    def recording_time(self):
+        """Return the total duration of all recording slices in seconds."""
+        if not self.recording_slices:
+            return 0
+        recording_time = sum(
+            (int(recording_slice.get("stop")) - int(recording_slice.get("start")))
+            for recording_slice in self.recording_slices
+            if recording_slice.get("stop") is not None
+        )
+        if self.is_recording:
+            recording_time += int(to_timestamp(timezone.now())) - int(
+                self.recording_slices[-1].get("start")
+            )
+        return recording_time
+
+    def _start_recording_slice(self, start):
+        """Start a new recording slice."""
+        slices = self.recording_slices
+        slices.append({"start": start})
+        self.recording_slices = slices
+        self.save()
+
+    def _stop_recording_slice(self, stop):
+        """Stop a recording slice"""
+        slices = self.recording_slices
+        slices[-1].update({"stop": stop, "status": PENDING})
+        self.recording_slices = slices
+        self.save()
+
+    def start_recording(self):
+        """Start recording the video."""
+        if not self.allow_recording:
+            raise VideoRecordingError("Recording is not allowed for this video.")
+
+        if self.is_recording:
+            raise VideoRecordingError("Video recording is already started.")
+
+        self._start_recording_slice(start=to_timestamp(timezone.now()))
```

Comment:
You are encapsulating too much thing in the model. It's not obvious what you are doing. Doing a `save` inside a method in the model is IMO a bad idea. The model is going to be a god object and we are loosing control on it. We must decouple this by introducing new concept maybe but I don't like to see this in the model

Reply:
So, I should remove all `VideoRecordingError`, and `save()` from model ?
For now, I think I will move them to `api/video.py`, is that ok ?

Reply:
I think you should create a [service layer](https://martinfowler.com/eaaCatalog/serviceLayer.html) and do it in this layer. Then you can use this layer in both the view and your command. You did that because you need to do the same thing in different places.
So I suggest to create a new repo `marsha/core/services` and create in it a `video_recording.py` module containing `start_recording` and `stop_recoring` methods. Both methods have as parameter a `video` model.